src/wallet/walletdb.cpp

```diff
@@ -471,6 +483,19 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
             // so set the wallet birthday to the beginning of time.
             pwallet->nTimeFirstKey = 1;
         }
+        else if (strType == "vkey")
```

Comment:
I wasn't sure about that; the watch addresses aren't there. IIUC the `IsKeyType` function is used to try and protect spending keys in the wallet in case of detected corruption.